### PR TITLE
SONARJAVA-1158 Brackets and Annotations for array dimensions

### PR DESCRIPTION
**Pull request by m-g-sonar**
Created new tree `ArrayDimensionTree`

### CODE REVIEW

File: java-squid/src/test/java/org/sonar/java/model/JavaTreeModelTest.java [hunk 1]
```diff
@@ -1950,9 +2033,14 @@ public void array_access_expression() {
...
     ArrayAccessExpressionTree tree = (ArrayAccessExpressionTree) expressionOfReturnStatement(code);
     assertThat(tree.is(Tree.Kind.ARRAY_ACCESS_EXPRESSION)).isTrue();
+    assertThatChildrenIteratorHasSize(tree, 2);
     assertThat(tree.expression()).isNotNull();
-    assertThat(tree.index()).isNotNull();
-    assertThatChildrenIteratorHasSize(tree, 4);
+    ArrayDimensionTree dimension = tree.dimension();
+    assertThat(dimension).isNotNull();
+    assertThat(dimension.openBracketToken().text()).isEqualTo("[");
+    assertThat(dimension.expression().is(Tree.Kind.INT_LITERAL));
```
Comment (SonarTech) [inline java-squid/src/test/java/org/sonar/java/model/JavaTreeModelTest.java]: CRITICAL Complete the assertion. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2970)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* CRITICAL 1 critical

Watch the comments in this conversation to review them.

File: java-squid/src/test/java/org/sonar/java/model/JavaTreeModelTest.java (hunk 1 above)
Comment (m-g-sonar) [inline java-squid/src/test/java/org/sonar/java/model/JavaTreeModelTest.java]: I love that rule!

**Comment by benzonico** [comment]
LGTM

**Pull request merged**